source/common/common/logger.h

```diff
@@ -323,6 +324,10 @@ template <Id id> class Loggable {
 
 /**
  * Convenience macro to get logger.
+ *
+ *(Note: Ensure you either use ENVOY_LOG_MISC(debug, FORMATSTRING, ...) for ad-hoc debug logs or
+ * have your class inherit from Logger::Loggable<Logger::Id::some_logger_id> to ensure
+ * __log_do_not_use_read_comment() does not cause a macro expansion error due to not being defined.)
```

Comment:
Optional: I think this could be shorter: Only use in methods of classes derived from Logger::Loggable. In other places, use ENVOY_LOG_MISC.